softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -470,12 +470,47 @@ func resourceSoftLayerVirtualGuestCreate(d *schema.ResourceData, meta interface{
 
 	log.Println("[INFO] Creating virtual machine")
 
-	guest, err := service.CreateObject(&opts)
-	if err != nil {
-		return fmt.Errorf("Error creating virtual guest: %s", err)
+	var id int
+
+	if opts.BlockDevices != nil && opts.BlockDeviceTemplateGroup != nil {
+		bd := *opts.BlockDeviceTemplateGroup
+		opts.BlockDeviceTemplateGroup = nil
+		opts.OperatingSystemReferenceCode = sl.String("UBUNTU_LATEST")
+		template, err := service.GenerateOrderTemplate(&opts)
+		if err != nil {
+			return fmt.Errorf("Error generating order template: %s", err)
+		}
+
+		// Remove temporary OS from actual order
+		for i, p := range template.Prices {
+			if strings.Contains(*p.Item.Description, "Ubuntu") {
+				template.Prices = append(template.Prices[:i], template.Prices[i+1:]...)
```

Comment:
this continual stitching of the array may result in a high-ish number of memory allocations.

what if you create a separate price array separately that has all the prices whose item description do not contain the world ubuntu. then you set it `template.Prices = ubuntuLessPricess`